### PR TITLE
Release 1.0

### PR DESCRIPTION
**Pull request by boblangley**
Conencts to https://github.com/Particular/DotNetCoreLaunch/issues/27

Pull request to review, and merge, 1.0 for release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,2 @@
-# NServiceBus.SagaAudit
-
+# ServiceControl.Plugin.Nsb6.SagaAudit
 Enables audit of saga state changes for NServiceBus version 6.x endpoints to facilitate the SagaView state change diagram in ServiceInsight.
```
Comment (boblangley) [inline README.md]: Perhaps this file change should be removed from the branch?

File: src/Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      Project="$(UserProfile)\.nuget\packages\gitversiontask\$(GitVersionTaskVersion)\buildMultiTargeting\GitVersionTask.targets"
+      Condition="Exists('$(UserProfile)\.nuget\packages\gitversiontask\$(GitVersionTaskVersion)\buildMultiTargeting\GitVersionTask.targets')" />
+  </ImportGroup>
+  <Target
+    Name="FixUpVersion"
+    BeforeTargets="_GenerateRestoreProjectSpec"
+    DependsOnTargets="GetVersion"
+    Condition="'$(GitVersion_Task_targets_Imported)' == 'True'" />
+
+</Project>
```
Comment (boblangley) [inline src/Directory.Build.props]: Should we skip this and go straight to `Particular.Packaging`?

File: src/NServiceBus.SagaAudit.AcceptanceTests/When_a_message_is_handled_by_a_saga.cs [hunk 3]
```diff
@@ -0,0 +1,171 @@
+﻿namespace ServiceControl.Plugin.Nsb6.SagaAudit.AcceptanceTests
```
Comment (boblangley) [inline src/NServiceBus.SagaAudit.AcceptanceTests/When_a_message_is_handled_by_a_saga.cs]: This is nit-picky, but should we kill the `Nsb6` portion of the namespace wherever we are using it?

File: src/NServiceBus.SagaAudit.AcceptanceTests/EndpointTemplates/ContextAppender.cs [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        {
+            return this;
+        }
+
+        public ILog GetLogger(string name)
+        {
+            return this;
+        }
+
+        public bool IsDebugEnabled { get{return true;}}
```
Comment (WilliamBZA) [inline src/NServiceBus.SagaAudit.AcceptanceTests/EndpointTemplates/ContextAppender.cs]: Nitpick: Some weird spacing around here

File: src/NServiceBus.SagaAudit/CaptureSagaResultingMessagesBehavior.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+            }
+
+            if (context.DeliveryOptions is PublishOptions)
+            {
+                var sagaResultingMessage = new SagaChangeOutput
+                {
+                    ResultingMessageId = context.OutgoingMessage.Id,
+                    TimeSent = DateTimeExtensions.ToUtcDateTime(context.OutgoingMessage.Headers[Headers.TimeSent]),
+                    MessageType = logicalMessage.MessageType.ToString(),
+                    //TODO: Can we remove the DeliveryDelay and DeliveryAt here??
```
Comment (WilliamBZA) [inline src/NServiceBus.SagaAudit/CaptureSagaResultingMessagesBehavior.cs]: Can we?

File: src/NServiceBus.SagaAudit/CaptureSagaStateBehavior.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            sagaAudit.SagaType = saga.GetType().FullName;
+            sagaAudit.SagaState = sagaStateString;
+
+            AssignSagaStateChangeCausedByMessage(sagaAudit, context);
+            backend.Send(sagaAudit);
+        }
+
+        public SagaChangeInitiator BuildSagaChangeInitatorMessage(Dictionary<string, string> headers, string messageId, string messageType )
+        {
+
```
Comment (WilliamBZA) [inline src/NServiceBus.SagaAudit/CaptureSagaStateBehavior.cs]: Seems weird to have a leading blank line here

File: src/NServiceBus.SagaAudit/SagaAuditFeature.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        public SagaAuditFeature()
+        {
+            DependsOn<Sagas>();
+            RegisterStartupTask<ServiceControlBackendInitializer>();
+        }
+        
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            context.Settings.TryGet("NServiceBus.SagaAudit.Serialization", out Func<object, Dictionary<string, string>> customSagaEntitySerialization);
```
Comment (WilliamBZA) [inline src/NServiceBus.SagaAudit/SagaAuditFeature.cs]: Side note: I LOVE this new syntax.

Actual comment: Could the `TryGet` ever return false? What happens if `customSagaEntitySerialization` is null?

File: src/NServiceBus.SagaAudit/ServiceControlBackend.cs [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    using Unicast.Transport;
+    using ServiceControl.EndpointPlugin.Messages.SagaState;
+    using SimpleJson;
+
+    class ServiceControlBackend
+    {
+        public Address Destination { get; set; }
+        public Address LocalAddress { get; set; }
+
+        public  ServiceControlBackend(ISendMessages messageSender)
```
Comment (WilliamBZA) [inline src/NServiceBus.SagaAudit/ServiceControlBackend.cs]: Nitpick: Double space here

**Comment by SzymonPobiega** [comment]
@boblangley I removed all references to the old namespace

**Comment by SzymonPobiega** [comment]
@boblangley I fixed the namespaces and package description. Should be clean now.

File: src/NServiceBus.SagaAudit/SagaEntitySerializationStrategy.cs [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+                return true;
+            }
+            output = null;
+            return false;
+        }
+        static bool IsSerializableType(Type t)
+        {
+            return t.IsPrimitive || t == typeof(string) || t == typeof(Guid) || t == typeof(TimeSpan) || t == typeof(DateTime) ||
+                   (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Nullable<>));
+        }
```
Comment (boblangley) [inline src/NServiceBus.SagaAudit/SagaEntitySerializationStrategy.cs]: @SzymonPobiega I just noticed the test you added to make sure nested types are _not_ serialized. Is this a change in behavior from the current plugins? Looking at them it looks like we just let the JSON serializer do it's thing, without restrictions.

**Comment by boblangley** [comment]
@SzymonPobiega My apologies for not posting this:

When we use Particular.Packaging and also explicitly bring in GitVersionTask as a dependency we do not need to do any of the workarounds. I rebased the branch and removed those commits and adjusted 2.0/3.0 accordingly.

I smoke tested the beta with ServiceInsight. All of our tests and acceptance tests are passing. I am merging this PR and staging 1.0 in octopus.

**Pull request merged**